src/NHibernate.Test/Linq/WhereSubqueryTests.cs

```diff
@@ -553,6 +553,28 @@ public void OrdersWithSubquery9A()
 			Assert.That(orderLines.Count, Is.EqualTo(711), nameof(orderLines));
 		}
 
+		[Test]
+		public void OrdersWithSubquery9Count()
+		{
+			if (Dialect is MySQLDialect)
+				Assert.Ignore("MySQL does not support LIMIT in subqueries.");
+
+			if (!Dialect.SupportsScalarSubSelects)
+				Assert.Ignore(Dialect.GetType().Name + " does not support scalar sub-queries");
+
+			var ordersQuery = db.Orders
+								.Where(x => x.Employee.EmployeeId > 5)
+								.OrderByDescending(x => x.OrderId)
+								.Take(2);
+
+			var orderLines = db.OrderLines
+									.Where(x => ordersQuery.Count(o => o == x.Order) > 0)
```

Comment:
In fact it could be anything. Count, sum, avg, etc.

Reply:
Please add a test `.Where(x => ordersQuery.Where(o => o == x.Order).Sum(o => o.OrderId) > 0)`

Reply:
Yeah.  I forgot to mention it. I tried to add aggregate functions but they don't work properly.

Reply:
For instance sum is generated along with TOP statement:
```C#
var orderLines = db.OrderLines
					.Where(x => ordersQuery.Sum(o => o.Freight.Value) > 0)
					.OrderBy(x => x.Id)
					.ToList();

```

Will give:
```SQL
where (select TOP (@p0) (sum(order1_.Freight)) from Orders order1_ where order1_.EmployeeId>@p1 order by order1_.OrderId desc)>@p2 order by orderline0_.OrderLineId asc]
```

Reply:
Hm.. Though with additional `Where(o => o == x.Order)` as in your suggested test it works properly.. So OK I will add it

Reply:
Done